File: app/src/main/java/com/savvasdalkitsis/gameframe/ip/usecase/IpDiscoveryUseCase.java
package com.savvasdalkitsis.gameframe.ip.usecase;

import com.savvasdalkitsis.gameframe.ip.model.IpAddress;

import rx.Observable;
import rx.subjects.BehaviorSubject;

public class IpDiscoveryUseCase {

    private BehaviorSubject<IpAddress> subject = BehaviorSubject.create();

    public Observable<IpAddress> monitoredIps() {
        return subject;
    }

    public void emitMonitoredAddress(IpAddress ip) {
        subject.onNext(ip);
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/ip/usecase/IpDiscoveryUseCase.kt
package com.savvasdalkitsis.gameframe.ip.usecase

import com.savvasdalkitsis.gameframe.ip.model.IpAddress
import io.reactivex.Flowable
import io.reactivex.processors.BehaviorProcessor

class IpDiscoveryUseCase {

    private val processor = BehaviorProcessor.create<IpAddress>()

    fun monitoredIps(): Flowable<IpAddress> = processor

    fun emitMonitoredAddress(ip: IpAddress) {
        processor.onNext(ip)
    }
}
